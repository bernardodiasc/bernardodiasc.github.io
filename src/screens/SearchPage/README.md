# SearchPage

## How to use

```
import SearchPage from 'screens/SearchPage'
```

```
<SearchPage />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
