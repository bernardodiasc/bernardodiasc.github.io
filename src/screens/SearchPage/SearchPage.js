import React, { Component } from 'react'

import Meta from 'components/Meta'
import LayoutContainer from 'containers/LayoutContainer'
import AppLink from 'displays/AppLink'
import TextBlock from 'displays/TextBlock'

class SearchPage extends Component {
  static defaultProps = {
    allCategories: [],
    allTags: [],
  }

  render() {
    const { allCategories, allTags } = this.props
    return (
      <LayoutContainer>
        <Meta
          title="Filter content"
          description=""
        />
        <TextBlock>
          To do:<br/>
          make the stuff below works<br/>
          and also make look better<br/>
          and also include a search input<br/>

          {allCategories.length > 0 && (
            <nav>
              <h3>Categories</h3>
              <ul>
                {allCategories.map(category => (
                  <li key={category}>
                    <AppLink to={`category/${category}`}>{category}</AppLink>
                  </li>
                ))}
              </ul>
            </nav>
          )}

          {allTags.length > 0 && (
            <nav>
              <h3>Tags</h3>
              <ul>
                {allTags.map(tag => (
                  <li key={tag}>
                    <AppLink to={`search/?${tag}`}>{tag}</AppLink>
                  </li>
                ))}
              </ul>
            </nav>
          )}
        </TextBlock>
      </LayoutContainer>
    )
  }
}

export default SearchPage
