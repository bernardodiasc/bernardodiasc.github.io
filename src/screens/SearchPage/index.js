import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
  getAllPostsCategories,
  getAllPostsTags,
} from 'selectors/data'

import SearchPage from './SearchPage'

const mapStateToProps = store => ({
  allCategories: getAllPostsCategories(store),
  allTags: getAllPostsTags(store),
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
  }, dispatch)
)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchPage)
