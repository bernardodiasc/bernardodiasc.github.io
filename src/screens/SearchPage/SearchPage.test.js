import React from 'react'
import { shallow } from 'enzyme'

import SearchPage from './SearchPage.js'

it('renders without crashing', () => {
  const component = shallow(<SearchPage />)
  expect(component).toMatchSnapshot()
})
