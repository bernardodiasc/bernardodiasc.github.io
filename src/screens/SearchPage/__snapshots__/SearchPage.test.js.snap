// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Connect(LayoutContainer)>
  <Meta
    description=""
    title="Filter content"
  />
  <TextBlock>
    To do:
    <br />
    make the stuff below works
    <br />
    and also make look better
    <br />
    and also include a search input
    <br />
  </TextBlock>
</Connect(LayoutContainer)>
`;
